content/docs/learning-guide/tutorials/environments/generate-terrain.md

```diff
@@ -0,0 +1,303 @@
+---
+linkTitle: Generate Terrain
+title: Generate Terrain with a Heightmap
+description: Learn to generate terrain in Open 3D Engine (O3DE) by creating entities and with Landscape Canvas.
+weight: 300
+toc: true
+---
+
+Terrain in Open 3D Engine (O3DE) is generated from *gradients*. A gradient, in this context, is a representation of changes in elevation. The gradient values range from dark (low elevation) to light (high elevation). Gradients can be generated procedurally or processed from an image. In this tutorial, you'll generate terrain with an image gradient, apply materials to the terrain, and add support for PhysX simulations with a heightfield collider and physics materials.
+
+This tutorial requires that you have the **Terrain**, **Landscape Canvas**, and **PhysX** Gems enabled in your project.
+
+## Required image asset
+
+To complete this tutorial, you need a *heightmap* that's similar to the following example:
+
+{{< image-width src="/images/learning-guide/tutorials/environments/heightmap-example.png" width="480" alt="An example heightmap image." >}}
+
+In a heightmap, dark values are low elevation areas and light values are high elevation areas. To use a heightmap for terrain in O3DE, it must be processed as a *gradient signal image (GSI)* by [Asset Processor](/docs/user-guide/assets/asset-processor/). To ensure the heightmap is processed correctly, keep the following recommendations in mind:
+
+* 16-bit color depths are recommended. 8-bit images may not provide enough gradient steps to represent elevation, and 32-bit images might be unnecessarily large. 16-bit images can provide up to 65,536 elevation steps.
+
+* Ensure the image is square with a power of two resolution (for example, 256 x 256, 512 x 512, 1024 x 1024, and so on). For the best result, each pixel in the heightmap should represent one world meter. A 1024 x 1024 heightmap, for example, represents just over one square kilometer of terrain.
```

Comment:
I added the bit about square power of 2 resolutions because I have run into issues processing textures in other contexts when I used arbitrary resolutions. I was also concerned about mobile development and issues with some lesser mobile gpus that might only support square power of 2 textures. Is saying "for the best result" better?

rewrite below:

* For the best result, prefer to use power of two resolutions (for example, 256, 512, 1024, and so on). The default **Heigh Query Resolution** for terrain in O3DE is 1.0 meters. This means a 1024 x 1024 heightmap represents just over one square kilometer of terrain. The **Height Query Resolution** can be changed in the **Terrain World** level component.

Reply:
So, even though we're using the StreamingImage texture asset format, the textures don't get uploaded to the GPU when used by Image Gradients. They're purely kept in main memory and pixels are read out of them by the CPU. We used to use a completely custom format for Image Gradients, but switched to this format a few months ago so that we could support mipmaps, reuse the textures in other contexts (which _might_ upload to the GPU), etc. But since we keep them CPU-side, it really shouldn't matter even for mobile what resolution they are.

I'd prefer not to encourage people towards power of two, because that could cause them to waste a lot of memory that's unnecessary. One of the benefits that people should be able to get from the current terrain design is that they can build and assemble assets that more directly match the size of their terrain. Instead of a single large 1024 x 1024 square, they can make a few chunks that are 800 x 180, 200 x 159, whatever, and arrange them around to make their playable surface. They shouldn't be forced to author (or load) areas of data that aren't going to be used or seen.